org.civicrm.reqangular/src/common/services/file.service.js

```diff
@@ -0,0 +1,34 @@
+/* globals URL */
+/* eslint-env amd */
+
+define([
+  'common/modules/services'
+], function (services) {
+  'use strict';
+
+  services.factory('fileService', ['$http', '$window', '$q', '$log', function ($http, $window, $q, $log) {
+    $log.debug('Service: fileService');
+
+    return {
+
+      /**
+       * Gets the file from server, creates its blob and opens it in new browser tab.
+       * @param  {object} file
+       * @return {object}
+       */
+      openFile: function (file) {
+        var deferred = $q.defer();
```

Comment:
Do you need $q here at all? $http returns promise https://docs.angularjs.org/api/ng/service/$http#get